### PR TITLE
Package coq-vcfloat.2.1

### PR DESCRIPTION
**Pull request by andrew-appel**
I wasn't able to follow the [instructions](https://coq.inria.fr/opam-packaging.html) exactly, because my fresh clone of this fork does not have an "upstream" repo in the way that the instructions specify, and I am not expert enough in git to understand how to adjust.  But I expect this form of P.R. should still work.

### CODE REVIEW

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
+opam-version: "2.0"
+version: "2.1"
```
Comment (palmskog) [inline released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:2]:
```suggestion
```
Version is defined by the filename path.

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+]
+install: [
+  [make "-C" "vcfloat" "-j%{jobs}%" "install" "INSTALLDIR=%{lib}%/coq/user-contrib/vcfloat"]
+]
+run-test: [
+  [make "-C" "vcfloat" "-j%{jobs}%" "tests"]
+]
+depends: [
+  "coq" {>= "8.16" & < "8.18~"}
+  "coq-flocq" {>= "4.1.0" & < "5.0"}
```
Comment (palmskog) [inline released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:31]:
```suggestion
  "coq-flocq" {>= "4.1.0" & < "5.0"}
  "coq-coquelicot"
```
I see the following in the CI log:
```
# coqc -Q . vcfloat IEEE754_extra.v
# File "./RAux.v", line 37, characters 0-42:
# Error: Cannot find a physical path bound to logical path
# Coquelicot with prefix Coquelicot.
```
Do you maybe have a bound on least version of Coquelicot as well?

**Review by palmskog — COMMENTED** [review]
Looks reasonable, but we are missing a dependency.

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam (hunk 2 above)
Comment (andrew-appel) [reply released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:31]: The dependency on Coquelicot is unnecessary.  I'll commit a new version of the vcfloat without it, then I'll adjust this P.R.

Comment (andrew-appel) [reply released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:30]: Done.  I hope nobody is offended that I moved the tag 2.1 on the VeriNum/vcfloat repository to a new commit, but now this PR to opam-coq-archive should build without any dependency on coquelicot.

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam (hunk 1 above)
Comment (andrew-appel) [reply released/packages/coq-vcfloat/coq-vcfloat.2.1/opam]: done, in latest commit

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+]
+url {
+  src: "https://github.com/VeriNum/vcfloat/archive/refs/tags/2.1.tar.gz"
+  checksum: "sha256=d9242c62a403b34282393708be8fcd23fb7387aae24bbe05de15463bb1ea2edd"
```
Comment (palmskog) [inline released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:34]:
```suggestion
  checksum: "sha256=d243d58667a55d76cd1e3727fa6701dbf0d78681a7b4d0301e83060fac056437"
```

File: released/packages/coq-vcfloat/coq-vcfloat.2.1/opam [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+]
+install: [
+  [make "-C" "vcfloat" "-j%{jobs}%" "install" "INSTALLDIR=%{lib}%/coq/user-contrib/vcfloat"]
+]
+run-test: [
+  [make "-C" "vcfloat" "-j%{jobs}%" "tests"]
+]
+depends: [
+  "coq" {>= "8.16" & < "8.18~"}
+  "coq-flocq" {>= "4.1.0" & < "5.0"}
```
Comment (palmskog) [inline released/packages/coq-vcfloat/coq-vcfloat.2.1/opam:30]:
```suggestion
  "coq-flocq" {>= "4.1.0" & < "5.0"}
  "coq-bignums"
```

**Comment by andrew-appel** [comment]
I see that the CI has failed, but in reviewing the output scripts of (e.g., 4.11.2) I don't actually see what the error was.

**Comment by andrew-appel** [comment]
Yes, coq-interval is a dependency.  I will update the PR.  But coq-compcert should not be a dependency, at most it should be used for the tests.

**Comment by andrew-appel** [comment]
Oops, it looks like VCFloat depends on a couple of things from the open-source part of CompCert.  It should be possible to eliminate that dependency for users of VCFloat who are not users of CompCert (and there certainly can be such applications), but for now I'll just put a compcert dependency in the opam file.

**Comment by andrew-appel** [comment]
But I have a basic opam question:  I tested my "opam install" of vcfloat locally on my machine, in an opam switch for the Coq Platform 2022.09 with the coq 8.16.1 package pick.  How can I get opam to tell me about all my dependencies, so that they don't pop up on the CI here?

**Comment by palmskog** [comment]
> How can I get opam to tell me about all my dependencies, so that they don't pop up on the CI here?

I don't think there's any good simple way besides creating a completely fresh opam switch and carefully and step-by-step installing all packages to make a project build with `make`. For example:

```shell
opam switch create 4.14.1+flambda+coq.8.17 --package=ocaml-variants.4.14.1+options,ocaml-option-flambda
eval $(opam env --switch=4.14.1+flambda+coq.8.17)
opam repo add coq-released https://coq.inria.fr/opam/released
opam pin add coq 8.17.0
```
.. then start the `make`/`opam install` cycle.

**Comment by palmskog** [comment]
> Oops, it looks like VCFloat depends on a couple of things from the open-source part of CompCert. It should be possible to eliminate that dependency for users of VCFloat who are not users of CompCert (and there certainly can be such applications), but for now I'll just put a compcert dependency in the opam file.

What you may want to do is create a special opam new opam package with the CompCert-reliant code, e.g., `coq-vcfloat-compcert`, and then have that depend on both `coq-vcfloat` and `coq-compcert`. The package would need a special `make` target, but can live in the same Git repository (a simple form of monorepo).

**Comment by palmskog** [comment]
@andrew-appel everything builds now, but inside the opam-invoked installation, it looks like `INSTALLFILES` variable from the vcfloat makefile is not being set property, which means the installation chokes on an `install` command.

**Comment by palmskog** [comment]
@andrew-appel `logpath` is a piece of metadata unique to the Coq opam archive, which as you suggest represents the Coq logical path (or paths) for the package content inside Coq. It has essentially two purposes: (1) informative for people reading package metadata and (2) the possibility of in the future suggesting packages to install when some dependency is missing inside Coq. We don't have any tools that do the latter yet, to my knowledge.

To be fully correct, `logpath` of the package should indeed be changed to `vcfloat` instead of `VCFloat`. I can do this if you want. But at the moment, it's a very minor issue due to lack of tools using `logpath`.

**Pull request merged**